socket: add support for -async (#229)
msteveb/jimtcl: jim_tcl.txt

is closed. Use `close -nodelete` in the rare case where this behaviour
should be avoided (e.g. after `os.fork`).

If '+-async+' is specified, the socket is opened in non-blocking (ndelay) mode
and 'connect' is non-blocking (applies to 'stream'). In this case, a 'writable' handler

> should be used that will fire when the connect succeeds or fails. In this case, 'peername'
> will succeed if connected or fail if connect failed. Typical usage is.

Reviewer: ```suggestion
should be used that will fire when the connect succeeds or fails, and 'peername'
will succeed if connected or fail if connect failed. Typical usage is as follows:
```

> should be used that will fire when the connect succeeds or fails. In this case, 'peername'
> will succeed if connected or fail if connect failed. Typical usage is.

Reviewer: ```suggestion
should be used that will fire when the connect succeeds or fails, and calling 'peername' on the handle
will succeed if connected or fail if connect failed. Typical usage is as follows:
```
